src/events/mod.rs

```diff
@@ -833,6 +846,49 @@ impl<'a> arbitrary::Arbitrary<'a> for BytesCData<'a> {
     }
 }
 
+/// Iterator over `CDATA` sections in a string.
+///
+/// This iterator is created by the [`BytesCData::escaped`] method.
+///
+/// If a string contains `]]>`, it needs to be split into multiple `CDATA`
+/// sections, splitting the `]]` and `>` characters. This iterator yields
+/// each of those sections.
+pub struct CDataIterator<'a> {
+    content: &'a str,
+    iter: MatchIndices<'a, &'a str>,
+    start: usize,
+    finished: bool,
+}
+
+impl<'a> Iterator for CDataIterator<'a> {
+    type Item = BytesCData<'a>;
+
+    fn next(&mut self) -> Option<BytesCData<'a>> {
+        if self.finished {
+            return None;
+        }
+
+        // If there are no more `]]>`, yield the remaining text and
+        // mark the iterator as finished.
+        let Some((next, _)) = self.iter.next() else {
```

Comment:
I think, that regular `match` would be more readable here. From my point of view `if let else` construction is hard to understand